bookmarks/utils.py

```diff
@@ -128,10 +129,16 @@ def redirect_with_query(request, redirect_url):
     return HttpResponseRedirect(redirect_url)
 
 
-def generate_username(email):
+def generate_username(email, claims):
     # taken from mozilla-django-oidc docs :)
-
     # Using Python 3 and Django 1.11+, usernames can contain alphanumeric
     # (ascii and unicode), _, @, +, . and - characters. So we normalize
     # it and slice at 150 characters.
-    return unicodedata.normalize("NFKC", email)[:150]
+    default_username = unicodedata.normalize("NFKC", email)[:150]
+    if settings.OIDC_USERNAME_CLAIM in claims:
+        return (
+            claims[settings.OIDC_USERNAME_CLAIM]
```

Comment:
Should this be normalized as well? Reading the comment above it seems there are limitations what characters the username can contain.

Reply:
That is right, i didn't think about that, while most OIDC providers already normalize this, we are never too safe.